speaker/bgp_controller.go

```diff
@@ -169,6 +170,13 @@ func (c *bgpController) ShouldAnnounce(l log.Logger, name string, _ []net.IP, po
 		return "nodeNetworkUnavailable"
 	}
 
+	if v, ok := nodes[c.myNode]; ok {
+		if v.Spec.Unschedulable {
```

Comment:
I'd extend the IsNetworkUnavailable function to some more generic name and embed any "this node is not good" logic into that.